superset-embedded-sdk/src/index.ts

```diff
@@ -143,6 +144,14 @@ export async function embedDashboard({
   ourPort.emit('guestToken', { guestToken });
   log('sent guest token');
 
+  async function refreshGuestToken() {
+    const newGuestToken = await fetchGuestToken();
+    ourPort.emit('guestToken', { guestToken: newGuestToken });
+    setTimeout(refreshGuestToken, getGuestTokenRefreshTiming(newGuestToken));
```

Comment:
I will take a look on this, but it does not create a recursion that will consume memory on every new timeout? It will free the calling function?

Reply:
No, there is not a memory leak. Calling `setTimeout` schedules a new function call, and the current frame is cleaned up.